provider/qsd.py

```diff
@@ -297,11 +297,12 @@ def _fulfil_image_props(self, name, params):
             LOG_JOB.info("QSD skip to create image %s ", name)
         else:
             # Record the images are maintained by QSD.
-            LOG_JOB.info("QSD ready to create image %s", name)
             obj = qemu_storage.QemuImg(params, data_dir.get_data_dir(), name)
-            _, result = obj.create(params)
-            if result.exit_status != 0:
-                raise QsdError("Failed create image %s " % name)
+            if params.get("force_create_image", "yes") == "yes":
```

Comment:
No need to modify qsd.py here. 
qsd has already handled the force_create_image, please see line291-294.
Please check the comments for images in cfg to see the root cause of creating stg1 twice.

Reply:
For the mirror/snapshot images, we never let qsd create them, so what about introducing a qsd_image_created param here? when we set it to 'yes', we just skip it, e.g. 
        if params.get_boolean('qsd_image_created'):
            image_created = True
        elif name in params.get("images").split():
             xxxxxx
So this flag param can avoid much update in params, e.g. without this new params, we need to set the following:
params['images'] += ' sn1'
params['create_image_sn1'] = yes
to avoid qsd to create sn1